Some improvements I made.

BrettRyland opened the pull request:
Assuming you're still maintaining this, here's a few improvements.
- Fix the epilepsy inducing shaking when the targeted vessel dies.
- Adjustable lerp rate in the settings.
- Lerp from previous position/orientation instead of resetting (due to the camera transform parent being set to null) in dogfight mode.
- Various small tweaks.
- Organisation of the code with `#region` `#endregion` tags.
- Auto-enable option for BDArmory competition.

Not done yet:
- Avoid stationary camera angles that are under ground/water for flying targets or really far away.

BrettRyland committed:
Fix 'Only custom filters can be played.' warnings due to AudioSource having 'playOnAwake' enabled by default in Unity.

BrettRyland committed:
Comment out debugging info.

BrettRyland commented:
Hey @jrodrigv, any chance you could merge this soon and make a new release. The only people benefiting from the improved DLL are those on the RWP/Munway/FC discord servers whom happened to pick up the DLL when I posted it there.

BrettRyland committed:
Smoother transitions and avoid jitter when switching targets and vessels.

BrettRyland committed:
Remove noDogfightTargetTimer as it doesn't seem to trigger now.

BrettRyland committed:
Reduce flickering by compensating for floating origin changes. Add on-screen debugging.

BrettRyland committed:
Correct grouping of debugging checks.

BrettRyland committed:
Fix NRE in atmospheric audio when vessel is destroyed.

BrettRyland committed:
Add roll parameter to dogfight camera.

BrettRyland committed:
Update changelog and bump version number.

BrettRyland committed:
Fix missing config file load/save bug.

BrettRyland committed:
Update changelog and version number.

BrettRyland committed:
Don't touch camera when we're not in control of it. Use TimeWarp instead of Time for velocity calculations. Allow switching into and out of IVA and Map modes without breaking.

BrettRyland committed:
Change the keypad move/zoom speeds to sliders for easier access. Disable slider discretisation when keypad control is enabled.

BrettRyland committed:
Properly correct for time-warp under almost all conditions for dogfight mode.

josuenos committed:
Merge pull request #6 from BrettRyland/master

Master

BrettRyland committed:
Add some more debugging.

BrettRyland committed:
Adjust DEBUG2 output

BrettRyland committed:
Migrate non-static config files to PluginData to avoid invalidating the ModuleManager cache (should give slightly faster start-up times). Exclude music volume levels from volumes based on distance from the camera. Adjust DEBUG2 debugging statements.

BrettRyland committed:
Update changelog, bump version number.

BrettRyland committed:
Try to reacquire a dogfight target if reverted due to a null dogfight target and Auto-Enable for BDArmory is enabled. Limit the type of vessels atmospheric audio gets added to to avoid saturating KSP's SoundManager channels unnecessarily.

BrettRyland committed:
Add sliders for configuring chances of 'random mode'. Add IVA view. Fix indexing bug in 'pathing mode' (note: the example path is still weird).

BrettRyland committed:
Changelog and spelling.

BrettRyland committed:
Improve the positioning of the Stationary Camera when the target is close to the ground or about to crash.

BrettRyland committed:
Don't restart camera on flight mode change when spawning.

BrettRyland committed:
Bump version number

BrettRyland committed:
Fix NRE when vessel is destroyed before switching to IVA mode. Don't immediately re-enter IVA mode if manually switched out of it.

BrettRyland committed:
Update the version number

BrettRyland committed:
Cached engine modules to avoid frequent GC allocations.

BrettRyland committed:
Trigger random mode selection on manual activation when random mode is on. Exit IVA mode more reliably. Optimisations to avoid GC allocations.

BrettRyland committed:
Update changelog, bump version number.

BrettRyland committed:
Adjust default randomMode chances.

BrettRyland committed:
Fix potential NRE. Don't add atmospheric controllers to objects that already have one.

BrettRyland committed:
Update changelog

BrettRyland committed:
Reduce log noise.

BrettRyland committed:
Abort trying to enter IVA mode if it throws an exception.

BrettRyland committed:
Update changelog, bump version number.

BrettRyland committed:
Recompile for KSP 1.12. Bump version.

BrettRyland committed:
Bump version number in AssemblyVersion too

BrettRyland committed:
Only activate the camera when changing modes if it was already active.

BrettRyland committed:
Switch stationary and pathing cameras to run in Update instead of FixedUpdate. Switch pathing camera to use unscaled time so that it works while paused. Don't automatically start playing a path when adjusting it or creating a new one and stop it if keyframes or the path is deleted. Adjust ordering of stationary path updates to avoid jitter.

BrettRyland committed:
Start playing a path when it's loaded if camera tools is active.

BrettRyland committed:
Tweak GUI layout.

BrettRyland committed:
Automatically disable TimeControl's camera zoom fix as it breaks CameraTools when slow-mo is enabled.

BrettRyland committed:
Update changelog, bump version number.

BrettRyland committed:
A variety of improvements

 - Add keybinding options for free move keys.
 - Rework stationary camera to actually make the camera stationary when below maxRelV.
 - Option to save the previously used rotation in stationary camera mode when no target is selected.
 - Add option of cubic spline interpolation to pathing mode positions based on https://en.wikipedia.org/wiki/Cubic_Hermite_spline#Interpolation_on_an_arbitrary_interval.
 - Add option of Slerp and cubic spline interpolation to pathing mode rotations.

BrettRyland committed:
Allow 'escape' to cancel keybinding

BrettRyland committed:
Fix incorrect adjustment of stationary camera position to allow manual movement again. Add atmospheric audio to the excluded list for doppler adjustments. Update changelog.

BrettRyland committed:
Add a try-catch around disabling CameraZoomFix in TimeControl in case something goes wrong.

BrettRyland committed:
Add note about piece-wise interpolation of paths.

BrettRyland committed:
Update changelog.

videopilot commented:
Any chance you want to add one more? Would be cool if there was an auto-select secondary target option. Maybe with a max-distance and option to randomly set to on?

BrettRyland commented:
@videopilot, the actively maintained branch of CameraTools is currently my one. The forum site is https://forum.kerbalspaceprogram.com/index.php?/topic/201063-camera-tools-continued-v1150/
If you're using BDArmory, then the "BDA AI Auto target" button will auto-select the secondary target once planes/vessels are targeting each other. If you're not using BDArmory, then there's no real way to decide which vessel, if any, should be the secondary target.

BrettRyland committed:
Make the menu toggle hotkey configurable too.

BrettRyland committed:
Add text input fields instead of sliders option.

BrettRyland committed:
Add in 'Maintain Vel.' equivalent of the old 'Initial Velocity' reference mode.

BrettRyland committed:
Make loading of paths a bit more fault tolerant.

BrettRyland committed:
Update changelog and bump version number.

BrettRyland committed:
Add customisable min/max values to the keypad move and zoom speeds (editable in the settings.cfg).

BrettRyland committed:
Update changelog, bump version number.

BrettRyland committed:
Re-add the 'interpolation rate' parameter in pathing. Update changelog. Bump version.

BrettRyland committed:
Add missing value setter for maxRelV.

Halbann committed:
QOL Tweaks

- Keys created during path playback are now automatically inserted at the current path time. Useful for molding paths in to specific shapes.

- Home/End on the numpad now moves the pathing camera up/down relative to the frame instead of the ground. Hold Left Alt to move relative to the ground. Allows for finer control than using the scroll wheel.

- Scroll wheel while using the pathing camera now adjusts move speed. Useful for quickly switching between fine adjustments and repositioning of the camera.

- The last path you used is now remembered between restarts.

- The GUI no longer closes on pressing save. Useful for habitual save pressers.

Halbann committed:
QOL Tweaks 2

- Added a 'Maintain Speed' button to the keyframe editor that adjusts the duration of the current keyframe to approximately match the speed between the previous two keyframes. Useful for smoothly extending paths and correcting speed inconsistencies in complex paths.

- New keys created while camera tools is inactive are now correctly started at the current camera position, rather than on top of the vessel facing down, or inside the vessel.

- The default time increment for new keys has been changed back to 10 seconds from 1 second. This is a more practical default path length and makes manually inserting new keys easier.

- The default rotation interpolation mode is now CubicSpline, was previously Slerp.

- Interpolation rate setting renamed to 'Camera Inertia' and the displayed value inverted to try to give the user a more intuitive understanding of what the setting does.

- Added a bit of feedback for when the user saves in pathing mode.

- Path playback now starts from the selected keyframe if one is selected. Useful for previewing small changes to long paths.

- The flight camera now correctly reverts to its original distance, orientation and mode when pathing deactivated.

Halbann committed:
QOL Tweaks 2

Halbann committed:
QOL Tweaks 2

Halbann committed:
QOL Tweaks 3

- Camera updates can now be switched between Update or 'Real-time' and FixedUpdate or 'In-Game Time'. 'Real-time' creates smoother paths and works while the game is paused, which helps to reduce lag, while 'In-Game Time' is useful for making sure pathing speed is consistent with game physics and that none of the path is skipped when processing footage to remove lag. See FFmpeg's mpdecimate.

- Camera zoom is now updated immediately rather than gradually when selecting a keyframe or starting a path.

BrettRyland committed:
Auto-Enable for BDArmory when using surface AIs too.

BrettRyland committed:
Merge branch 'dev' into pathing-QoL

BrettRyland committed:
Adjustment to quality of life tweaks in PR#4

- Modifier NUM0 default changed to "right shift" — right shift works well as a modifier key (it's used as the modifier for accessing the cheat menu in Linux).
- Default hotkey [/] changed to [0] — BDArmory now uses the [/] key for the AI GUI, so [0] is a better default hotkey for toggling the menu.
- Camera Inertia — The variable needs to be this way (instead of 1-lerp) to show precision when close to 0. I've reversed the direction of the slider to be more natural. Regarding naming, "Camera Inertia" implies some actual inertia of the camera, which isn't what's happening, it's just a Lerp to the target position, which is different, so I've changed this to "Camera Lerp Rate".
- Adjusted width of "Bind Key" to be slightly narrower and better positioning of "Press a Key".
- Toggling the menu with the hotkey now saves the settings too.
- Real-time vs In-Game time: UpdateStationaryCamera shouldn't depend on the useRealTime variable, which is specific to the Pathing mode.
- Add parsing of int type to CTPersistantField to avoid conversion to and from float for logically integer variables.
- flightCameraTransform manipulation is wrong in the creation of new keyframe — fixed to use StartPathingCam instead, which handles this.
- position vs localPosition in orig transforms — local is relative to the parent transform, which may have been modified (or even removed) while we were messing with the camera, so it's better to use the absolute position/rotation.

BrettRyland committed:
Save and restore camera mode too.

BrettRyland committed:
Set the camera mode directly instead of using 'setModeImmediate' to avoid the annoying camera mode change messages.

BrettRyland committed:
Change the default modifier key for pathing movement to 'keypad enter' for better ergonomics. Replace path 'lerpRate' with 'secondarySmoothing' to get a linear scale from 0 to 1 (lerp rate is calculated internally as 10^(-2*secondarySmoothing)). Automatically convert lerpRate in old paths to secondarySmoothing. Adjust GUI formatting to fit 'Secondary Smoothing:' label.

BrettRyland committed:
Save and restore using the local position and rotation if the original parent is non-null. Set the camera parent correctly when creating a new keyframe.

Halbann committed:
QOL Tweaks 4

- Tweaked modified movement to move relative to ship orientation instead of the ground; similar functionality when landed and more useful in orbit. 

- Tweaked new key creation to take stock zoom if it is currently adjusted and camera tools is deactivated. Useful for creating a path entirely via moving the stock flight camera.

- Added restoration of stock zoom when camera tools is deactivated.

Halbann committed:
QOL Tweaks 4

BrettRyland committed:
Apply the modified movement to mouse movement too. Add in up/down motion (middle+left buttons) since scrolling which previously did this was changed. Slight optimisation.

BrettRyland committed:
Reverse forwardAxis and rightAxis when viewing vessel from in front.

BrettRyland committed:
Update changelog. Bump version number.

BrettRyland committed:
Use the reversed 'forwardAxis' and 'rightAxis' all the time to avoid awkwardness when transitioning from behind to infront of the vessel.

BrettRyland committed:
Update changelog

BrettRyland committed:
Add autoEnableOverride for BDArmory's tournament inter-round time-warping.

BrettRyland committed:
Add comments on inability of adding doppler.

BrettRyland committed:
Fix random mode transitions that got broken in the last release. Reset the zoom slider when resetting the FOV. Auto-switch the auto-zoom to regular zoom when entering pathing mode as pathing only uses the regular zoom. Remove the temporaryRevert code (I think that was a Scott Manley hack). Delay activating the new camera mode on vessel switches until LateUpdate for all camera modes, not just dogfight mode.

BrettRyland committed:
Update changelog

BrettRyland committed:
Check for the flight camera being null on startup.

BrettRyland committed:
Allow the stationary camera to be placed up to 5km higher to avoid terrain line-of-sight issues.

BrettRyland committed:
Ignore maxRelV when random mode is enabled.

josuenos committed:
Merge branch 'BrettRyland:dev' into dev

BrettRyland committed:
Fix check for secondary camera target priorities.

BrettRyland committed:
Bump version number

BrettRyland committed:
Separate zoom and autozoom parameters for the different modes and don't reset the zoom value when reverting the FoV.

BrettRyland committed:
Update changelog

BrettRyland committed:
Remove 0 minimum for maxRelV to allow reverse fly-bys.

josuenos committed:
Merge branch 'BrettRyland:dev' into dev

BrettRyland committed:
Checks for null flight camera.

BrettRyland committed:
Fix an NRE in the audio controller.

BrettRyland committed:
Use signed maxRelVSqr to calculate the initial offsets for auto-flybys.

BrettRyland committed:
Corrections to the KrakensbaneWarpCorrection for dogfight mode.

BrettRyland committed:
Fix the lower limit of the camera shake multiplier when using numeric fields. Tweak the camera shake slider to use steps of 0.1.

BrettRyland committed:
Make the config paths relative to the KSP app location.

BrettRyland committed:
Add a version number and activation toggle/indicator to the GUI.

BrettRyland committed:
Only disable TimeControl's CameraZoomFix while CameraTools is active. Also, look for and disable BetterTimeWarp's ScaleCameraSpeed while CameraTools is active, since that also breaks CameraTools during slow-mo.

BrettRyland committed:
BDA Auto Targeting: add an option to not target incoming missiles.

BrettRyland committed:
Also ignore the underAttack field when checking for BDA incoming missiles to ignore.

BrettRyland committed:
Corrections to the KrakensbaneWarpCorrection for dogfight and stationary camera modes so that they work (almost) correctly at all altitudes and warp levels.

BrettRyland committed:
Fix typo in changelog

josuenos committed:
Merge branch 'BrettRyland:dev' into dev

josuenos committed:
Add VTOL AI to BDA AI checks

josuenos committed:
Merge branch 'dev' of https://github.com/josuenos/CameraTools into dev

BrettRyland committed:
Updated fields/properties for refactor in BDArmory. Don't revert the camera when there's no further dogfight targets.

BrettRyland committed:
Merge branch 'dev' into HEAD

BrettRyland committed:
Reword entry in changelog

BrettRyland committed:
Add check for the class type of VesselSpawner due to the current changes in BDA dev.

BrettRyland committed:
Lower the log error for not being able to set IVA camera mode to a warning.

SuicidalInsanity committed:
Centroid Mode

Add new Camera option to Dogfight Mode to follow centroid of the furball instead of just target ship

BrettRyland committed:
Merge branch 'dev' into HEAD

BrettRyland committed:
Fix csproj paths and merge latest dev.

BrettRyland committed:
Add GetBDVessels function to fetch the active vessels and teams from BDArmory.
Collect the BDArmory fields/properties into its own region.

BrettRyland committed:
Merge branch 'dev' into pr7

BrettRyland committed:
Clean up the 'UpdateLoadedVessels' function slightly.

BrettRyland committed:
Replace reflection for BDArmory fields/properties with delegates for much faster access. Fix BindingFlags for initial reflection due to changes in BDArmory.

BrettRyland committed:
Include ReflectionUtils.cs too.

BrettRyland committed:
Add note about speed-up factor of using delegates instead of reflection.

BrettRyland committed:
Update the changelog

BrettRyland committed:
Bump version to 1.22.0 for release.

BrettRyland committed:
Fix some memory leaks detected by KSPCF.

BrettRyland committed:
Refactor integration with other mods into their own files (mostly). Some BDArmory-related settings may need resetting.

BrettRyland committed:
Allow deploying to multiple KSP instances when compiling in Linux.

BrettRyland committed:
Add speed-mode input for keyboard input.

BrettRyland committed:
Add display field-width parameter to numeric input fields. Update numeric input fields when making changes with keyboard input. Disable speed free-move mode when numeric input mode is enabled.

BrettRyland committed:
Cache the atmospheric audio sound clips to avoid GC allocations. Initialise the 'maxRelV' numeric field properly so that the field-width parameter is 6, not the min value.

BrettRyland committed:
Auto-landing camera. Some fixes for AppleSilicon being buggy in this version of Unity.

BrettRyland committed:
Update changelog and bump version number.

BrettRyland committed:
AudioSource debugging. Save and restore the other fields that got adjusted too.

BrettRyland committed:
Inertial chase mode for dogfight camera.

BrettRyland committed:
Rework the BDA secondary target priority to favour the vessel's target (instead of attacker) when it has recently been firing or fired a missile (requires the latest BDA+). Add a slider to control the minimum update interval of the BDA secondary target selection after switching targets (updates occur every 0.5s after the minimum interval has elapsed). Standardise many of the GUI Rects.

BrettRyland committed:
Update changelog with specific BDA+ version number.

BrettRyland committed:
Prevent the camera from going below the surface in dogfight mode when the vessel is landed or near the surface when splashed.

BrettRyland committed:
Update README

BrettRyland committed:
Ignore ksp_dir.txt lines starting with # for post-build events in Linux

BrettRyland committed:
Disable roll and offset in dogfight mode when the vessel is an EVA kerbal.

BrettRyland committed:
Update changelog

BrettRyland committed:
Fix bug in field name for input boxes. Add DEBUG2 info for camera altitude.

BrettRyland committed:
Update changelog and bump version number for release.

BrettRyland committed:
Fix dogfight centroid mode.

BrettRyland committed:
Add checks for the active vessel being a BDA missile and use dogfight chase mode if it is. Add roll option to stationary camera using right+middle mouse buttons, with the movement modifier key (keypad enter) switching between camera and world coordinate systems.

BrettRyland committed:
Add support for using the MouseAimFlight in dogfight camera mode. Optimise access to flight camera transform in dogfight camera mode.

BrettRyland committed:
Delay camera activation/deactivation when in map mode until back in flight mode.

BrettRyland committed:
Bump version numbers for release.

BrettRyland committed:
Add delayed activation too.

BrettRyland committed:
Update changelog

BrettRyland committed:
Adjust random mode selection for low altitude to be more amenable to pod-racing.

BrettRyland committed:
Add a visual toggle for free-move mode (speed vs position).

BrettRyland committed:
Add an optional keybind for resetting the roll in stationary camera mode. Initially unbound, remove the entry from the settings.cfg to unbind it once bound.

BrettRyland committed:
Use empty string, not null for unassigned key.

BrettRyland committed:
Fix camera position in Stationary and Dogfight modes when returning from Map mode.

BrettRyland committed:
Add a free-look mode to Dogfight mode (hold right mouse button).

BrettRyland committed:
Bump version number for release.

BrettRyland committed:
Add a configurable delay to activating free-look mode (default: 0.1s).

BrettRyland committed:
Add a configurable movement threshold to activating free-look mode (default: 0.1).

BrettRyland committed:
If the camera parent gets stolen, automatically steal it back if it's the original camera parent (due to spawning a kerbal, BDA MML, etc.).

BrettRyland committed:
Allow customising the maximum value of the zoom, auto zoom, dogfight distance and dogfight offsets in the settings.cfg.

BrettRyland committed:
Clean up setting/resetting audio source fields.

BrettRyland committed:
Apply some Roslyn suggested clean-up.

BrettRyland committed:
Perform a check for a spatializer and disable doppler effects if none is found.

BrettRyland committed:
Update changelog

BrettRyland committed:
Do nothing with part audio until the original settings are stored.

BrettRyland committed:
Also in ApplyEffects

BrettRyland committed:
Rework the sonic boom calculations, allowing them to reset and removing the booming when switching vessels/restarting camera modes.

BrettRyland committed:
Add some more optimisation and a missing volume factor for wind tear.

BrettRyland committed:
Bump version numbers for release.

BrettRyland committed:
Also always add part audio controllers since they get removed when OnResetCTools triggers.

BrettRyland committed:
Fix wind-tear audio being overly loud from previous tweaks.

BrettRyland committed:
Fix wind-tear audio being overly loud from recent tweaks.

BrettRyland committed:
Fix an AudioSource memory leak.

BrettRyland committed:
Add changelog entry. Bump version number for bugfix release.

BrettRyland committed:
Fix aero FX not being applied and provide a toggle to disable them. Clean up DEBUG2 messages.

BrettRyland committed:
Add check for BDOrbitalAI.

BrettRyland committed:
Adjustments to fix some issues when BDArmory should be inhibiting CameraTools and more detailed debugging messages.

BrettRyland committed:
Bump version number for release.

BrettRyland committed:
Add missing changelog entry

BrettRyland committed:
Sanitise timestamps by shifting duplicates by 0.001s instead of breaking the view frustrum with NaNs and avoid division by 0 in spline calculations.

BrettRyland committed:
Bump version number for bugfix release.

BrettRyland committed:
Fix NREs from atmospheric audiosources when not in atmosphere.

BrettRyland committed:
Minor tweaks to MouseAimFlight helpers.

BrettRyland committed:
Don't reset the original camera parent's position when reverting (fixes weird camera view).

BrettRyland committed:
Add support for a new property in BDA for inhibiting camera tools.

BrettRyland committed:
Override the camera restore distance with custom distance (from BDA) when reverting due to auto-enabling for BDA.

BrettRyland committed:
Adjust debug log tags for BDA mod-integration.

BrettRyland committed:
Add support for new BDA helpers for tracking missiles' targets in dogfight mode.
Some internal restructuring of utils.
Begin adding localisation support.
Begin adding tooltips.

BrettRyland committed:
When missiles have no target vessel or position, revert to velocity chase mode properly.

BrettRyland committed:
Update changelog. Bump version number for release.

BrettRyland committed:
Improvements to death camera angles for dogfight mode.

BrettRyland committed:
Add a button to switch from pathing mode to stationary mode while maintaining the current view.
Adjust location of Random mode options.
Add a button for resetting the camera roll (in stationary and pathing modes).
Optimisations to use vessel.CoM instead of vessel.transform.position where applicable.
Apply maxRelV to the stationary camera when using Random mode.
Finish adding English (en-us) localisation and tooltips.

BrettRyland committed:
Add note about splitting tooltips over multiple lines.

BrettRyland committed:
Move the auto-zoom toggle above the slider.

BrettRyland committed:
Adjust default settings for a dynamic dogfight mode.

BrettRyland committed:
Fix stationary death cam.

BrettRyland committed:
Separate the auto-zoom toggle between dogfight and stationary modes.

BrettRyland committed:
Be consistent with the death cam transform.

BrettRyland committed:
Fix death camera transitions. Improve camera placement for vessels near surface.

BrettRyland committed:
Add a slider for setting the minimum interval to check for a new secondary dogfight target if the secondary target is a missile. Don't restart the dogfight camera when switching secondary targets for smoother transitions.

BrettRyland committed:
Reduce unnecessary revert and restarts.

BrettRyland committed:
Krakensbane optimisations. Fix deathcam rotation issue.

BrettRyland committed:
Include the KrakensbaneUtils.cs file

BrettRyland committed:
Bump version numbers for release.

BrettRyland committed:
Add a slider for controlling how long after BDA's secondary dogfight target dies (for non-missiles) before selecting a new secondary non-missile target (to enable lingering on kills).

BrettRyland committed:
Adjust some slider lengths for better formatting.

BrettRyland committed:
Fix false positives of Krakensbane being active on vessel switches.

BrettRyland committed:
Bump version number for release

BrettRyland committed:
Add version tag to distribution zip file.

ThreeMACH committed:
Add files via upload

Localization - Simplified Chinese

BrettRyland committed:
Add changelog entry for Chinese localisation

BrettRyland committed:
Add an option to toggle pivot points between the camera and the target vessel when rotating with the right mouse button in pathing mode. When using the movement modifier in pathing mode, rotation with the right mouse button is now relative to the target vessel's orientation instead of the camera's orientation.

BrettRyland committed:
Add the corresponding pivot modes for stationary and dogfight cameras.
Fix camera roll when transitioning between modified and non-modified movement modes.
The free-move mode (position vs speed) and pivot mode are now preserved in the settings.
Increase the mouse sensitivity for free-look in dogfight mode.

BrettRyland committed:
Respect the targetCoM option in stationary mode. In dogfight mode, pivot around the closest point to the vessel along the camera's forward axis.

BrettRyland committed:
Tweak mouse input scales.

BrettRyland committed:
Tweak changelog and tooltip for the pivot toggle

BrettRyland committed:
Bump version number for release

BrettRyland committed:
Add better handling of dogfight mode for BDA waypoint mode with combat. Adjust get/set accessor templates to return the specified types instead of generic objects.

BrettRyland committed:
Update changelog

BrettRyland committed:
Reduce unintended roll due to lerping in dogfight mode. Add to-do notes.

BrettRyland committed:
Add a slider for selecting the reference frame to use for the dogfight offset.

BrettRyland committed:
Adjust the dogfight inertial chase slider to the range 0—1 and adjust the 'inertia' behaviour, including affecting camera rotation.

BrettRyland committed:
Don't switch modes on vessel switch if random mode is not enabled.

BrettRyland committed:
Fix the unintended roll due to lerping. Add chase-plane mode. Lerp the roll in inertial and chase-plane modes. .transform optimisations.

BrettRyland committed:
Add option to choose which part the chase plane mode tracks.

BrettRyland committed:
Add controller input (AXIS_CAMERA_HDG and AXIS_CAMERA_PITCH) to freelook in dogfight mode and camera rotation in stationary and pathing modes.

BrettRyland committed:
Add a geospatial pathing option to pathing mode where points are GPS coordinates and rotations are absolute instead of relative to the active vessel.

BrettRyland committed:
Bump version number for release.

BrettRyland committed:
Add a UI scaling option that is tied to the global UI_SCALE setting by default.

BrettRyland committed:
Tweak some debugging.

BrettRyland committed:
Bump version numbers for release.

BrettRyland committed:
Add in a missing Krakensbane correction for the dogfight inertial chase camera rotation target.

BrettRyland committed:
Bump version numbers for bugfix release.

BrettRyland committed:
Use vessel.ReferenceTransform instead of vessel.transform in the dogfight camera for when the controlling part isn't the root part and may have a different orientation.

BrettRyland committed:
Fall back to the regular vessel.transform in case the reference transform is null.

BrettRyland committed:
Use vessel.vesselTransform as the fallback and update it each frame in case the reference transform becomes null due to lost parts. Do the same for the stationary camera.

BrettRyland committed:
Update changelog

BrettRyland committed:
Bump version numbers for bugfix release.

BrettRyland committed:
Reset lerp momentum for the dogfight inertial chase mode when starting the dogfight camera.

BrettRyland committed:
Revert to pivoting around the camera in stationary mode if in target-pivot mode and there's no target to pivot around.

BrettRyland committed:
Move random mode overrides into ChooseRandomMode so that it's usable by other callers. Respect random mode when auto-starting for BDA.

BrettRyland committed:
Adjust autoFlyBy/randomMode offsets in stationary camera to account for vessel size.

BrettRyland committed:
Include vessel radius when checking the random mode override.

BrettRyland committed:
Add changelog entries.

BrettRyland committed:
Use vessel.transform.position instead of vessel.CoM for checking LoS from the initial stationary camera position due to weird parts allowing the CoM to be below terrain.

BrettRyland committed:
Bump version for release

BrettRyland committed:
Minor layout tweaks to account for strings being clipped for some UI scaling values.

BrettRyland committed:
Include the UI scaling for the tooltip position (can't adjust size though).

BrettRyland committed:
Compatibility fixes with pre-KSP v1.12 versions.

BrettRyland committed:
Add some changelog entries and bump the version number for a bugfix release.
